neuro_flow/batch_runner.py

```diff
@@ -152,15 +155,24 @@ async def process(
                     deps = ctx.graph[tid]
                     needs = {}
                     for dep_id in deps:
-                        dep = finished.get(dep_id)
-                        assert dep is not None
-                        needs[dep_id] = DepCtx(dep.status, dep.outputs)
+                        if dep_id in skipped:
+                            needs[dep_id] = DepCtx(JobStatus.CANCELLED, {})
```

Comment:
Not sure if the skipped task should have CANCELLED state

Reply:
I agree, I'll introduce DISABLED task state.